src/main/java/com/google/sps/data/Folder.java

```diff
@@ -89,25 +83,23 @@ public static void deleteFolder(Entity folder) throws IOException {
     }
   }
 
-  public static void deleteFolderWithRetries(Entity folder) {
-    int retries = 5;
-    while (true) {
+  public static void deleteFolderWithRetries(Entity folder, int retries) {
```

Comment:
You can use overloading to create a default parameter value:

```java
static const int DEFAULT_NUM_RETRIES = 5;
public static void deleteFolderWithRetries(Entity folder) {
  return deleteFolderWithRetries(folder, DEFAULT_NUM_RETRIES);
}
```

Reply:
done